pkg/alerts/types.go

```diff
@@ -0,0 +1,45 @@
+package alerts
+
+import (
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	"github.com/prometheus/common/model"
+)
+
+type MetricCondition struct {
+	AlertName      string
+	AlertNamespace string
+	AlertLevel     string
+
+	// Text is the description of why this alert condition matched.
+	Text string
+
+	Matches func(sample *model.Sample) bool
+}
+
+type MetricConditions []MetricCondition
+
+func (c MetricConditions) MatchesInterval(alertInterval monitorapi.Interval) *MetricCondition {
```

Comment:
Key fix was in here, the old version of this function was only partially implemented and thus our alert matching for allowed alerts hasn't been working for likely months.